SFML/SFML
# Add WindowBase::setImePreEditPosition

## Description

This PR adds a method to `WindowBase`, `setImePreEditPosition(const Vector2i& pos)`, that hints to the OS/IME framework where to put the preedit window.

## Tasks

* [x] Tested on Linux (Fcitx5)
* [x] Tested on Linux (IBus)
* [ ] Tested on macOS (Not implemented)

I'm completely illiterate when it comes to macOS/Objective-C (or any other Apple stuff), and I need help on that part.

## How to test this PR?

Install an IME if you don't have one, and run this example.

This is relatively straightforward on Windows 10: install a language of Chinese Simplified/Traditional or Japanese in the Settings app, and press Win+Space to switch to that input.

```cpp
#include <SFML/Graphics.hpp>

int main() {
    sf::RenderWindow window(sf::VideoMode({ 400, 400 }), "IME position");
    window.setFramerateLimit(60);

    while (window.isOpen()) {
        for (sf::Event event; window.pollEvent(event);) {
            if (event.type == sf::Event::Closed)
                window.close();
            else if (event.type == sf::Event::MouseButtonPressed) {
                auto pos = sf::Mouse::getPosition(window);
                window.setImePreEditPosition({pos.x, pos.y});  // Try negating these values and see what happens    
            }
        }

        window.clear();
        window.display();
    }
}
```
Click around in the opened window, and try typing with the IME engaged.
![图片](https://user-images.githubusercontent.com/19706363/227417968-405e73f3-8e7a-42e8-a3b5-0989efad9855.png)
